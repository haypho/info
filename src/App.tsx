import React from "react";
import { HashRouter, Route } from "react-router-dom";
import NavBar from "./components/nav-bar";
import Home from "./pages/Home";
import Projects from "./pages/Projects";

const App: React.FC = () => (
  <HashRouter basename="/">
    <NavBar />
    <Route exact path="/" component={Home} />
    <Route path="/projects" component={Projects} />
  </HashRouter>
);

export default App;
