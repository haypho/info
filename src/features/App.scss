@use '../styles/colors';
@use '../styles/sizes';

.app-header {
  background: colors.$background;
}

.app-header .container {
  text-align: center;
  padding: 1em 0;
  justify-content: space-between;
}

.app-title {
  margin-bottom: 0;
}

.app-subtitle {
  font-size: 0.75rem;
}

.app-content {
  flex-grow: 1;
}

@media (min-width: sizes.$breakpoint) {
  .app-header .container {
    flex-direction: row;
  }
}