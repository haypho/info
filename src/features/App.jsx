import './App.css';
import React from 'react';

const App = () => <div>Hello from main App. This is an addition.</div>;

export default App;
