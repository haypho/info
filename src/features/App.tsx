import './App.scss';
import React from 'react';

const App: React.FC = () => (
  <>
    <header className="app-header">
      <div className="app-title-area">
        <h2 className="app-title">Hayden Phothong</h2>
        <p className="app-subtitle">PERSONAL PORTFOLIO</p>
      </div>
    </header>
    <img src="/assets/images/profile.jpg" alt="profile" />
    <footer className="app-footer">
      <p>Hayden Phothong &copy; 2020</p>
    </footer>
  </>
);

export default App;
