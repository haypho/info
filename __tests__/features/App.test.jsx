import React from 'react';
import { render } from '@testing-library/react';
import App from '../../src/features/App';

describe('when rendering the App', () => {
  it('should match the snapshot', () => {
    const sut = render(<App />);

    expect(sut.asFragment()).toMatchSnapshot();
  });
});
