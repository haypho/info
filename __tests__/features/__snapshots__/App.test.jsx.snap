// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering the App should match the snapshot 1`] = `
<DocumentFragment>
  <a
    href="#/"
  >
    Home
  </a>
  <a
    href="#/projects"
  >
    Projects
  </a>
  <h1>
    Welcome!
  </h1>
</DocumentFragment>
`;
